### ISSUE TITLE
Create wpiformat GitHub Action

### ISSUE THREAD

**Issue by Starlight220** [wpilibsuite/styleguide#189]
To sum up some discussion from wpilibsuite/allwpilib#2619:
Create & publish a wpiformat GH Action so teams can easily setup CI/CD in their repo with a syntax similar to:
```yml
- uses: wpilibsuite/check-wpiformat@master
  name: Check WPIFormat Output
```
This action should also be used by the relevant wpilibsuite repositories instead of the bulkier Action/Azure setup. 
The `clang-format.sh` script should also be moved to this repo, then run internally at the start of the action if needed.
Adding `.github/workflows/[find_a_good_name].yml` and relevant `.styleguide` files as part of the template project would be a good idea.

#### To sum up: Setting up a wpiformat CI/CD should be effortless for teams! 
If it runs without any team setup/intervention/editing - even better!

I can attempt to do most of this myself, but help and design decisions would be helpful.
If anyone else wants to try this, some relevant github docs articles:
- [Creating Actions](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions)
- [Publishing an Action](https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace#publishing-an-action)

**Comment by calcmogul** [comment]
wpiformat only helps C++ teams (who are a dwindling minority). Given the amount of manual setup work it requires, I'm not sold on its utility for the average team. You should make or find a https://github.com/diffplug/spotless GitHub Action for Java teams instead.

**Comment by Starlight220** [comment]
I didn't know about spotless. Is there a reason it isn't included in the template java project `build.gradle`?

**Comment by calcmogul** [comment]
We don't want a failed spotless run to prevent teams from deploying code.
